src/widgets/infinite-hits/infinite-hits.tsx

```diff
@@ -12,20 +14,58 @@ import {
 } from '../../lib/utils';
 import { component } from '../../lib/suit';
 import { withInsights, withInsightsListener } from '../../lib/insights';
+import {
+  WidgetFactory,
+  Template,
+  Hit,
+  InsightsClientWrapper,
+} from '../../types';
 
 const withUsage = createDocumentationMessageGenerator({
   name: 'infinite-hits',
 });
 const suit = component('InfiniteHits');
 const InfiniteHitsWithInsightsListener = withInsightsListener(InfiniteHits);
 
+export type InfiniteHitsCSSClasses = {
+  root: string | string[];
+  emptyRoot: string | string[];
+  item: string | string[];
+  list: string | string[];
+  loadPrevious: string | string[];
+  disabledLoadPrevious: string | string[];
+  loadMore: string | string[];
+  disabledLoadMore: string | string[];
+};
+
+export type InfiniteHitsTemplates = {
+  empty: Template<void>;
+  showPreviousText: Template<void>;
+  showMoreText: Template<void>;
+  item: Template<Hit>;
+};
+
+export type InfiniteHitsRendererWidgetParams = {
```

Comment:
Why do we have a type related to the renderer at this level? It should probably be into the connector. Those values are indeed required by the renderer but not by the widget itself. At the moment we can't use the widget without any of those options.